flytekit/common/tasks/task.py

```diff
@@ -268,6 +273,58 @@ def _python_std_input_map_to_literal_map(self, inputs):
             for k, v in _six.iteritems(self.interface.inputs)
         })
 
+    def _produce_deterministic_version(self, version=None):
+        """
+        :param Text version:
+        :return Text:
+        """
+
+        if self.container is not None:
+            raise ValueError("Client-side task versions can only be used for containerless tasks")
+        if version is not None:
+            return version
+        custom = _json_format.Parse(_json.dumps(self.custom, sort_keys=True), _struct.Struct()) if self.custom else None
+
+        # The task body is the entirety of the task template MINUS the identifier. The identifier is omitted because
+        # 1) this method is used to compute the version portion of the identifier and
+        # 2 ) the SDK will actually generate a unique name on every task instantiation which is not great for
+        # the reproducibility this method attempts.
+        task_body = (self.type, self.metadata.to_flyte_idl().SerializeToString(deterministic=True),
+                     self.interface.to_flyte_idl().SerializeToString(deterministic=True), custom)
+        return _hashlib.md5(str(task_body).encode('utf-8')).hexdigest()
+
+    @_exception_scopes.system_entry_point
+    def register_and_launch(self, project, domain, name=None, version=None, inputs=None):
+        """
+        :param Text project: The project in which to register and launch this task.
+        :param Text domain: The domain in which to register and launch this task.
+        :param Text name: The name to give this task.
+        :param Text version: The version in which to register this task
+
+        :param dict[Text, Any] inputs: A dictionary of Python standard inputs that will be type-checked, then compiled
+            to a LiteralMap.
+        """
```

Comment:
Add the return type?

Reply:
I want to start using python3 typing.

Reply:
done

Reply:
I think revamping everything to use Python 3 typing will be a separate effort - one that will probably take a while to do.  We can incrementally add along the way sure, either way, no preference.